src/main/java/com/compilercharisma/chameleonbusinessstudio/service/AuthenticationService.java

```diff
@@ -1,11 +1,17 @@
 package com.compilercharisma.chameleonbusinessstudio.service;
 
+import com.compilercharisma.chameleonbusinessstudio.entity.UserEntity;
 import com.compilercharisma.chameleonbusinessstudio.entity.user.AbstractUser;
 import com.compilercharisma.chameleonbusinessstudio.exception.NoUserLoggedInException;
 import com.compilercharisma.chameleonbusinessstudio.exception.UserNotRegisteredException;
+
+import reactor.core.publisher.Mono;
+
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.security.authentication.AnonymousAuthenticationToken;
 import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.ReactiveSecurityContextHolder;
+import org.springframework.security.core.context.SecurityContext;
 import org.springframework.security.core.context.SecurityContextHolder;
 import org.springframework.security.oauth2.core.user.OAuth2User;
 import org.springframework.stereotype.Service;
```

Comment:
Overall I think AuthenticationService is making it 100x harder than it has to be with Spring Security

Reply:
As per above, the reason I need this class is because our DTO isn't being set as the principal, so this handles mapping the OAuth2 user to our domain